### ISSUE TITLE
prefer-object-spread: also flag loops that should be spreads

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [palantir/tslint#3817]
### Bug Report

- __TSLint version__: 5.9.1
- __TypeScript version__: 2.8.1
- __Running TSLint via__: (pick one) VS Code

#### TypeScript code being linted

```ts
const alpha = { a: "a" };
const beta = { b: "b" };

const combined = {};

for (const i in alpha) {
    if ({}.hasOwnProperty.call(alpha, i)) {
        combined[i] = alpha[i];
    }
}

for (const i in beta) {
    combined[i] = beta[i];
}
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "prefer-object-spread": true
    }
}
```

#### Actual behavior

No errors.

#### Expected behavior

Looping over `alpha` with the same `hasOwnProperty` check spreading uses after declaring the variable as `const` should be flagged as a place where it'd be cleaner to use spread.

Looping over `beta` _could_ be flagged in the same manner, but is technically different without `hasOwnProperty` flags. Perhaps under a different flag?

For reference, TypeScript's `__assign` helper:

```javascript
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
```

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all`*`!

`*` _(not that anybody other than me participated in this issue... rip)_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
